core/templates/dev/head/pages/exploration_editor/editor_tab/StateEditor.js

```diff
@@ -336,21 +336,22 @@ oppia.factory('trainingDataService', [
         _removeAnswer(answer);
 
         var answerGroup = responsesService.getAnswerGroup(answerGroupIndex);
-        var rules = answerGroup.ruleSpecs;
+        var rules = answerGroup.rules;
 
         // Ensure the answer group has a classifier rule.
         var classifierRule = null;
         for (var i = 0; i < rules.length; i++) {
           var rule = rules[i];
-          if (rule.rule_type === CLASSIFIER_RULESPEC_STR) {
+          if (rule.ruleType === CLASSIFIER_RULE_STR) {
             classifierRule = rule;
             break;
           }
         }
         if (!classifierRule) {
           // Create new classifier rule for classification. All classifiers
           // should match this schema.
-          classifierRule = angular.copy(DEFAULT_CLASSIFIER_RULE_SPEC);
+          classifierRule = RuleObjectFactory.createFromDict(
```

Comment:
Why not bring this into the RuleObjectFactory? I.e. RuleObjectFactory.createNewClassifierRule()